packages/firebase/index.d.ts

```diff
@@ -4008,7 +4008,35 @@ declare namespace firebase.analytics {
      */
     logEvent(
       eventName: 'add_payment_info',
-      eventParams?: { [key: string]: any },
+      eventParams?: {
+        coupon?: EventParams['coupon'];
+        currency?: EventParams['currency'];
+        items?: EventParams['items'];
+        payment_type?: EventParams['payment_type'];
+        value?: EventParams['value'];
+        [key: string]: any;
+      },
+      options?: firebase.analytics.AnalyticsCallOptions
+    ): void;
+
+    /**
+     * Sends analytics event with given `eventParams`. This method
+     * automatically associates this logged event with this Firebase web
+     * app instance on this device.
+     * List of official event parameters can be found in
```

Comment:
Curious about the word "official."  Do we just mean "valid?"

I can see omitting an adjective entirely; "See <link> for a list of event parameters."

Reply:
No, now that I think about it, the term I've seen in other analytics docs might be "recommended".  Users can put any string they want and it will just be ingested as a custom value, but the "recommended"(?) ones are the ones that the analytics backend recognizes and can do some special things with. Is "recommended" the right word? It's actually usually used for event types, not event params: https://support.google.com/firebase/answer/6317498?hl=en&ref_topic=6317484

Reply:
Aha -- "recommended" definitely sounds best, if the value can be any string.

Reply:
Changed to "recommended".